### PR TITLE
Support ACH in Gift Entry Batch

### PR DESCRIPTION
**Pull request by npsp-reedestockton**
# Critical Changes

# Changes
- We've enabled ACH as a payment method in Batch Gift Entry.

# Issues Closed

# Community Ideas Delivered

# Features Intended for Future Release

# Features for Elevate Customers

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by npsp-reedestockton** [comment]
Tracking [W-11831824](https://gus.lightning.force.com/lightning/r/a07EE000017Xyt5YAC/view)

**Comment by npsp-reedestockton** [comment]
@lparrott @voduyemi @daniel-fuller Tagged all three of you as reviewers for this item. It includes some commits from #7105 and #7104. Added everything here to enable end to end testing.

File: force-app/main/default/classes/ElevateCreateBatchItemRequest.cls [hunk 1]
```diff
@@ -28,50 +28,72 @@
...
     }
 
     private void buildPurchaseInfoBody(ElevateCreateBatchItemRequestDTO batchItemRequestDTO,
-                                       PS_IntegrationServiceConfig.Service configService) {
+                                       PS_IntegrationServiceConfig.Service configService,
+                                       String gatewayOverride) {
         purchaseInfo = new ElevateCreateBatchItemRequest.PurchaseInfo();
         purchaseInfo.merchantId = configService.getMerchantIds();
-        purchaseInfo.gatewayId = configService.getGatewayIds();
+        if (String.isNotBlank(gatewayOverride)) {
```
Comment (lparrott) [inline force-app/main/default/classes/ElevateCreateBatchItemRequest.cls:61]: Might not be worth it, but could the `configService` be passed the `gatewayOverride` and then move the if/else logic into the service itself?

File: force-app/main/default/lwc/geGatewaySettings/geGatewaySettings.js [hunk 2]
```diff
@@ -1,20 +1,56 @@
...
+        if (this.elevateSettings?.uniqueKey) {
+            this.decryptElevateGateway().then(error => {
+                if (this.elevateSettings?.uniqueKey && !this.decryptedGatewayId) {
+                    showToast(psGatewayNotValid, details, 'error', 'sticky');
+                }
+            });
+        }
+    }
+
+    clearDecryptedElevateSettings() {
```
Comment (lparrott) [inline force-app/main/default/lwc/geGatewaySettings/geGatewaySettings.js:32]: Could this instead be called `initSingleGiftSettings` or something like that?

File: force-app/main/default/lwc/geFormWidgetTokenizeCard/geFormWidgetTokenizeCard.js [hunk 3]
```diff
@@ -282,19 +285,40 @@ export default class geFormWidgetTokenizeCard extends LightningElement {
...
         }
     }
 
+    isValidBatchElevatePaymentMethod() {
+        if (!(this.isPaymentMethodAch() || this.isPaymentMethodCreditCard())) {
+            return false;
+        }
+
+        if (!GeGatewaySettings.getElevateSettings()) {
+            return true;
```
Comment (lparrott) [inline force-app/main/default/lwc/geFormWidgetTokenizeCard/geFormWidgetTokenizeCard.js:310]: At first glance I'm not sure why this is a valid case... feels like we're just missing settings.

File: force-app/main/default/lwc/geGatewaySelectWidget/geGatewaySelectWidget.js [hunk 4]
```diff
@@ -21,18 +21,24 @@ import GeGatewaySettings from 'c/geGatewaySettings';
...
     @track isACHEnabled = true;
     @track isACHDisabled = false;
     @track isCreditCardEnabled = true;
     @track isCreditCardDisabled = false;
     @track isLoading = true;
     @track isDefaultTemplate = false;
     @track isGatewayAssignmentEnabled = false;
+    @track isGatewaySelectionDisabled = false;
+
+    @api parentId = '';
```
Comment (lparrott) [inline force-app/main/default/lwc/geGatewaySelectWidget/geGatewaySelectWidget.js:41]: Should this be set to blank if it's being passed in from a parent component? I don't think it needs to be initialized at all.

**Review by lparrott — COMMENTED** [review]
Looks good, just had some minor questions/suggestions!

File: force-app/main/default/lwc/geFormWidgetTokenizeCard/geFormWidgetTokenizeCard.js (hunk 3 above)
Comment (npsp-reedestockton) [reply force-app/main/default/lwc/geFormWidgetTokenizeCard/geFormWidgetTokenizeCard.js]: We are missing settings. Not every template will have elevateSettings. In that case, we do not disable any payment methods.

**Review by voduyemi — APPROVED** [review]
Thanks @npsp-reedestockton!

**Review by lparrott — APPROVED** [review]

**Pull request merged**